src/libfuncs/bytes31.rs

```diff
@@ -0,0 +1,238 @@
+    <TLibfunc as GenericLibfunc>::Concrete: LibfuncBuilder<TType, TLibfunc, Error = Error>,
+{
+    let value = &info.c;
+    let value_ty = registry.build_type(
+        context,
+        helper,
+        registry,
+        metadata,
+        &info.signature.branch_signatures[0].vars[0].ty,
+    )?;
+
+    let op0 = entry.append_operation(arith::constant(
+        context,
+        Attribute::parse(context, &format!("{value} : {value_ty}")).unwrap(),
+        location,
+    ));
+    entry.append_operation(helper.br(0, &[op0.result(0)?.into()], location));
+
+    Ok(())
+}
+
+/// Generate MLIR operations for the `bytes31_to_felt252` libfunc.
+pub fn build_to_felt252<'ctx, 'this, TType, TLibfunc>(
+    context: &'ctx Context,
+    registry: &ProgramRegistry<TType, TLibfunc>,
+    entry: &'this Block<'ctx>,
+    location: Location<'ctx>,
+    helper: &LibfuncHelper<'ctx, 'this>,
+    metadata: &mut MetadataStorage,
+    info: &SignatureOnlyConcreteLibfunc,
+) -> Result<()>
+where
+    TType: GenericType,
+    TLibfunc: GenericLibfunc,
+    <TType as GenericType>::Concrete: TypeBuilder<TType, TLibfunc, Error = CoreTypeBuilderError>,
+    <TLibfunc as GenericLibfunc>::Concrete: LibfuncBuilder<TType, TLibfunc, Error = Error>,
+{
+    let felt252_ty = registry.build_type(
+        context,
+        helper,
+        registry,
+        metadata,
+        &info.branch_signatures()[0].vars[0].ty,
+    )?;
+    let value: Value = entry.argument(0)?.into();
+
+    let op = entry.append_operation(arith::extui(value, felt252_ty, location));
+
+    let result = op.result(0)?.into();
+
+    entry.append_operation(helper.br(0, &[result], location));
+
+    Ok(())
+}
+
+/// Generate MLIR operations for the `u8_from_felt252` libfunc.
+pub fn build_from_felt252<'ctx, 'this, TType, TLibfunc>(
+    context: &'ctx Context,
+    registry: &ProgramRegistry<TType, TLibfunc>,
+    entry: &'this Block<'ctx>,
+    location: Location<'ctx>,
+    helper: &LibfuncHelper<'ctx, 'this>,
+    metadata: &mut MetadataStorage,
+    info: &SignatureOnlyConcreteLibfunc,
+) -> Result<()>
+where
+    TType: GenericType,
+    TLibfunc: GenericLibfunc,
+    <TType as GenericType>::Concrete: TypeBuilder<TType, TLibfunc, Error = CoreTypeBuilderError>,
+    <TLibfunc as GenericLibfunc>::Concrete: LibfuncBuilder<TType, TLibfunc, Error = Error>,
+{
+    let range_check: Value = entry.argument(0)?.into();
+    let value: Value = entry.argument(1)?.into();
+
+    let felt252_ty = registry.build_type(
+        context,
+        helper,
+        registry,
+        metadata,
+        &info.param_signatures()[1].ty,
+    )?;
+    let result_ty = registry.build_type(
+        context,
+        helper,
+        registry,
+        metadata,
+        &info.branch_signatures()[0].vars[1].ty,
+    )?;
+
+    let op = entry.append_operation(arith::constant(
+        context,
+        Attribute::parse(context, &format!("{} : {}", u8::MAX, felt252_ty)).unwrap(),
+        location,
+    ));
```

Comment:
Shouldn't the range be up to `2 ** 248 - 1` (or (`1 << (31 * 8)) - 1`)?

Reply:
true, i copy pasted it from u8 and forgot to change it